pkg/rehearse/jobs.go

```diff
@@ -266,37 +263,78 @@ func getResolvedConfigForTest(ciopConfigs config.DataByFilename, resolver regist
 // being also changed by the tested PR.
 func inlineCiOpConfig(container *v1.Container, ciopConfigs config.DataByFilename, resolver registry.Resolver, metadata api.Metadata, testname string, loggers Loggers) (apihelper.ImageStreamTagMap, error) {
 	allImageStreamTags := apihelper.ImageStreamTagMap{}
-	// replace all ConfigMapKeyRef mounts with inline config maps
-	for index := range container.Env {
-		env := &(container.Env[index])
-		if env.Name == "CONFIG_SPEC" {
-			// if CONFIG_SPEC has already been set, do not add new CONFIG_SPEC section
+	if container.Command == nil || container.Command[0] != "ci-operator" {
+		return allImageStreamTags, nil
+	}
+
+	var hasConfigEnv bool
+	var ciopConfig config.DataWithInfo
+	var envs []v1.EnvVar
+	for idx, env := range container.Env {
+		switch {
+		case env.Name == "CONFIG_SPEC" && env.ValueFrom != nil:
+			// job attempts to get CONFIG_SPEC from cluster resource, which is weird,
+			// unexpected and we cannot support rehearsals for that
+			return nil, fmt.Errorf("CONFIG_SPEC is set from a cluster resource, cannot rehearse such job")
+		case env.Name == "UNRESOLVED_CONFIG" && env.ValueFrom != nil:
+			// job attempts to get UNRESOLVED_CONFIG from cluster resource, which is weird,
+			// unexpected and we cannot support rehearsals for that
+			return nil, fmt.Errorf("UNRESOLVED_CONFIG is set from a cluster resource, cannot rehearse such job")
+		case env.Name == "CONFIG_SPEC" && env.Value != "":
+			// job already has inline CONFIG_SPEC: we should not modify it
 			return allImageStreamTags, nil
+		case env.Name == "UNRESOLVED_CONFIG" && env.Value != "":
+			if err := yaml.Unmarshal([]byte(env.Value), &ciopConfig.Configuration); err != nil {
+				return nil, fmt.Errorf("failed to unmarshal UNRESOLVED_CONFIG: %w", err)
+			}
+			// Annoying hack: UNRESOLVED_CONFIG means this is a handcrafted job, which means
+			// `testname` cannot be relied on (it is derived from job name, which is arbitrary
+			// in handcrafted jobs). We need the test name to know which `tests` field to
+			// resolve, so we try to detect it from `--target` arg, if present.
+			//
+			// The worst case is that we do not find the matching name. In such case,
```

Comment:
Wouldn't `ci-operator` run all tests when no `--target` is present?

Reply:
Hmm I thought it would just build all artifacts and run no tests. How much would we care about that? I'm a bit worried about hitting some content limit with full resolution, I suppose someone added that optimization for a reason.